Mathlib/LinearAlgebra/Matrix/Ideal.lean

```diff
@@ -113,141 +114,193 @@ end Ideal
 
 /-! ### Two-sided ideals in a matrix ring -/
 
+namespace RingCon
+variable {R n : Type*}
+
+section NonUnitalNonAssocSemiring
+variable [NonUnitalNonAssocSemiring R] [Fintype n]
+variable (n)
+
+/-- The ring congruence of matrices with entries related by `c`. -/
+def matricesOver (c : RingCon R) : RingCon (Matrix n n R) where
```

Comment:
I don't quite understand why the name wasn't `matrixOver` ...

Reply:
Or just `.matrix`?

Reply:
toMatrix ofMatrix seem to be the convention.

Reply:
I'm not sure I agree, for instance I'd expect `RingCon.pi` not `RingCon.toPi`

Reply:
Okay I agree with you, so we should use  .matrix and .ofMatrix.
I was thinking about OrderDual.toDual etc..

Reply:
Done for the RingCon API.